api/utils.go

```diff
@@ -103,6 +103,48 @@ func proxyAsync(engine *cluster.Engine, w http.ResponseWriter, r *http.Request,
 	r.URL.Host = engine.Addr
 
 	log.WithFields(log.Fields{"method": r.Method, "url": r.URL}).Debug("Proxy request")
+
+	var (
+		writeCompleted    = make(chan bool)
+		detectedCompleted = make(chan bool)
+	)
+
+	//if docker server did not write any data in body ( just like request /containers/xxx/stats to stopped container) , `io.Copy(NewWriteFlusher(w), resp.Body)` will wait for reading data.
+	//But the client that connect to swarm may close the request . So we should detect it and  cancel the request to docker server.
+	go func() {
+
+		var (
+			closeNotifier http.CloseNotifier
+			ok            bool
+		)
+
+		type canceler interface {
+			CancelRequest(*http.Request)
+		}
+
+		if closeNotifier, ok = w.(http.CloseNotifier); ok == false {
```

Comment:
usually I see `!ok` instead of `ok == false`. Style thing, but I think `!ok` is more idiomatic.